CMakeLists.txt

```diff
@@ -29,15 +31,32 @@ project(reflectcpp)
 
 set(CMAKE_CXX_STANDARD 20)
 
-if (REFLECTCPP_BUILD_SHARED)
-    add_library(reflectcpp SHARED src/yyjson.c)
-else ()
-    add_library(reflectcpp STATIC src/yyjson.c)
-endif ()
+if(REFLECTCPP_USE_BUNDLED_DEPENDENCIES)
+    if (REFLECTCPP_BUILD_SHARED)
+        add_library(reflectcpp SHARED src/yyjson.c)
+    else()
+        add_library(reflectcpp STATIC src/yyjson.c)
+    endif()
+else()
+    if (REFLECTCPP_BUILD_SHARED)
+        add_library(reflectcpp SHARED)
+    else()
+        add_library(reflectcpp STATIC)
+    endif()
+endif()
 
+set_target_properties(reflectcpp PROPERTIES LINKER_LANGUAGE CXX)
 target_compile_features(reflectcpp PUBLIC cxx_std_20)
 
-target_include_directories(reflectcpp PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include> )
+if(REFLECTCPP_USE_BUNDLED_DEPENDENCIES)
+    target_include_directories(reflectcpp PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include> $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/rfl/thirdparty>)
```

Comment:
If the `thirdparty` folder as set as extra include here, there should be no need to have the checks in the header files. I guess. Compiler should use the first header it finds, so if `thirdparty` is first, it should be okay.